src/library/scala/collection/immutable/RedBlackTree.scala

```diff
@@ -850,23 +850,23 @@ private[collection] object RedBlackTree {
     }
 
   private[this] def _union[A, B](t1: Tree[A, B], t2: Tree[A, B])(implicit ordering: Ordering[A]): Tree[A, B] =
-    if((t1 eq null) || (t1 eq t2)) t2
-    else if(t2 eq null) t1
+    if((t2 eq null) || (t2 eq t1)) t1
```

Comment:
no real change here on line 853/4 863/4 and I find the original easier to read

Reply:
Agreed. I good here as a result of an automated refactoring sequence, but I've changed them around now.